subsys/net/ip/tcp.c

```diff
@@ -661,12 +661,47 @@ int net_tcp_prepare_ack(struct net_tcp *tcp, const struct sockaddr *remote,
 	return -EINVAL;
 }
 
+static void get_src_addr_ptr(struct net_tcp *tcp,
```

Comment:
I don't think this is a right change, see https://github.com/zephyrproject-rtos/zephyr/issues/5481

Reply:
I tried to find helper function for this purpose. May be I missed it. I will take a look.